### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by GiuseppeTG**
#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
```
Comment (vechicin) [inline base_decorator.rb:7]: [OPTIONAL] As opposed to JavaScript, you don't need to use `super()` in your class constructor in Ruby. Feel free to remove it.

Screen Shot 2022-08-23 at 9 09 49

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = @nameable.correct_name
+    return name[0..9] if name.length > 10
+
+    name
+  end
```
Comment (vechicin) [inline trimmer_decorator.rb:9]: [OPTIONAL] You can implement this method in the following manner to save you some lines of code.

Screen Shot 2022-08-23 at 9 11 15

**Review by vechicin — APPROVED** [review]
# Hi @GiuseppeTG ✋, 

Your project is complete! 🎯 🧠 💯 
There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights ⚡ 

- [x] Good communication with the reviewer
- [x] Clean code
- [x] README file

## Optional suggestions 📂 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 2 above)
Comment (GiuseppeTG) [reply trimmer_decorator.rb:9]: What elegance the one of France 👍

**Pull request merged**